test-utils/fixtures/NRSBrokerFixtures.scala

```diff
@@ -0,0 +1,124 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fixtures
+
+import models.nrs._
+import models.response.nrsBroker.NRSBrokerInsertPayloadResponse
+import play.api.libs.json.{JsObject, JsValue, Json}
+import uk.gov.hmrc.auth.core.AffinityGroup.Organisation
+import uk.gov.hmrc.auth.core.ConfidenceLevel
+import uk.gov.hmrc.auth.core.retrieve._
+import utils.SHA256Hashing
+
+import java.nio.charset.StandardCharsets
+import java.time.Instant
+import java.util.Base64
+
+trait NRSBrokerFixtures extends BaseFixtures {
+
+  val identityDataModel: IdentityData = IdentityData(
+    internalId = Some(testInternalId),
+    externalId = Some("externalId"),
+    confidenceLevel = ConfidenceLevel.L200,
+    agentInformation = AgentInformation(Some("agentId"), Some("agentCode"), Some("agentFriendlyName")),
+    itmpName = None,
+    itmpAddress = None,
+    loginTimes = LoginTimes(Instant.ofEpochMilli(1L), None),
+    credentialRole = None,
+    affinityGroup = Some(Organisation),
+    credentials = Some(Credentials(providerId = testCredId, providerType = "gg"))
+  )
+
+  val predicateRetrieval = new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(new ~(
+    Some(Organisation),
+    Some(testInternalId)),
+    Some("externalId")),
+    None),
+    Some(Credentials(providerId = testCredId, providerType = "gg"))),
+    ConfidenceLevel.L200),
+    None),
+    None),
+    None),
+    None),
+    AgentInformation(Some("agentId"), Some("agentCode"), Some("agentFriendlyName"))),
+    None),
+    None),
+    None),
+    None),
+    None),
+    None),
+    LoginTimes(Instant.ofEpochMilli(1L), None)
+  )
+
+  val testJsonPayload: JsObject = Json.obj(
+    "testing" -> "emcs-tfe",
+    "version" -> "1"
+  )
+
+  val testPlainTextPayload: String =
+    """{
+      |    "testing": "emcs-tfe",
+      |    "version": "1"
+      |}""".stripMargin
+
+  val nrsMetadataModel: NRSMetadata = NRSMetadata(
+    businessId = "emcs",
+    notableEvent = "emcs-explain-a-shortage-ui",
+    payloadContentType = "application/json",
+    payloadSha256Checksum = "80298ad82661b0744d95cd969f782fdde6db73e92d25f52ef0b77b803bfbf4d9",
+    userSubmissionTimestamp = Instant.ofEpochMilli(1L),
+    identityData = identityDataModel,
+    userAuthToken = testAuthToken,
+    headerData = Json.obj("key" -> "value"),
+    searchKeys = SearchKeys(testErn)
+  )
+
+  val nrsMetadataJson: JsValue = Json.obj(
+    "businessId" -> "emcs",
+    "notableEvent" -> "emcs-explain-a-shortage-ui",
```

Comment:
should the notableEvent contain the word 'excess' aswell - as it can be a shortage or excess?

Reply:
This was what the event was defined as in `emcs-tfe` so I didn't want to deviate from what had already been signed off with NRS.

Reply:
This was the notable event from `emcs-tfe`:
- https://github.com/hmrc/emcs-tfe/commit/04ecc3e1da77ce8906d4bdbf01a2e92505452e1f#diff-e53883545d8b1623059c432b14e03e0cb13461aa04126a2b530dd1acc7ee1b0fL33